app/src/main/java/com/example/task/ToDoFragment.kt

```diff
@@ -16,26 +26,33 @@ class ToDoFragment : Fragment() {
 
     private val sharedViewModel: TaskViewModel by activityViewModels()
 
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var taskList: LiveData<List<Task>>
     private var _binding: FragmentTodoBinding? = null
-    // This property is only valid between onCreateView and
-    // onDestroyView.
     private val binding get() = _binding!!
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-
         _binding = FragmentTodoBinding.inflate(inflater, container, false)
         return binding.root
 
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
+                recyclerView = binding.recyclerViewForFragmentToDo
+                recyclerView.layoutManager = LinearLayoutManager(activity)
+                lifecycleScope.launch{
+                    TaskViewModel().getData().observe(viewLifecycleOwner,
+                        Observer {
+                            recyclerView.adapter = PendingAdapter(it)
+                        })
+                }
```

Comment:
Bad Indentation. Use Ctrl + Alt + I (eye) to format.

Reply:
Also, you don't really need to obtain a recyclerview instance right? Just do binding.recyclerViewForFragmentToDo everywhere, it's not that big.

Reply:
Would it not be too lengthy code that way? when every time i would write binding.recyclerViewForFragmentToDo. Anyways I have made the changes.